ztp/siteconfig-generator/siteConfig/siteConfigHelper.go

```diff
@@ -0,0 +1,44 @@
+package siteConfig
+
+import (
+	"log"
+
+	yaml "gopkg.in/yaml.v3"
+)
+
+// Add ztp deploy annotation to all siteconfig generated CRs
+// A dynamic value might be added later
+func addZTPAnnotation(clusterCRs []interface{}) ([]interface{}, error) {
+
+	for _, v := range clusterCRs {
+		cr := v.(map[string]interface{})
+		metadata := cr["metadata"].(map[string]interface{})
+		annotations, _ := metadata["annotations"].(map[string]interface{})
+		annotations[ztpAnnotation] = nil
+		if cr["kind"] == "ConfigMap" {
+			// Add ztp annotation to embedded machine config CRs
```

Comment:
I would much rather see us not special-case the "look at the contents of the ConfigMap" here, but instead call `addZTPAnnotation` on each of the individual MachineConfigs when we process them in siteConfigBuilder.

Reply:
Ok, addZTPAnnotationToManifest() is called on the individual  manifest.